sklearn/linear_model/cd_fast.pyx

```diff
@@ -218,9 +257,8 @@ def enet_coordinate_descent(np.ndarray[DOUBLE, ndim=1] w,
 
                 if w[ii] != 0.0:
                     # R -=  w[ii] * X[:,ii] # Update residual
-                    daxpy(n_samples, -w[ii],
-                          <DOUBLE*>(X.data + ii * n_samples * sizeof(DOUBLE)),
-                          1, <DOUBLE*>R.data, 1)
+                    axpy(n_samples, -w[ii], &X_data[ii * n_samples], 1,
```

Comment:
Grr... and somehow what's broken is the newly included `ATL_srefaxpy`? It looks entirely innocuous, i.e. it's doing the same except multiplying by the weight for each element in turn.

I'm curious whether the bug appears if you reduce the compiler optimisation level.